### PR TITLE
fix(frontend): multi-select-tag tag creation/selection fixes

### PR DESCRIPTION
**Pull request by yonmey**
**What is the problem this PR is trying to solve?**
- When selecting a value and typing again the same value, the creation of the same value was allowed.
- If a tag was set, for instance, `tag1` and we wanted to create a tag named `tag`, it didn't allow to do it as the regex that verifies the suggestions was evaluating true even with the first typed character.

**What is the chosen solution to this problem?**
I filtered the already used tags to not be able to create them again.
I added a new regex with the exact match.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js [hunk 1]
```diff
@@ -190,14 +190,19 @@ export default class MultiSelectTag extends React.Component {
...
+				.filter(item => !currentProps.value.includes(item.value));
 
 			if (currentValue) {
 				const escapedValue = escapeRegexCharacters(currentValue.trim());
-				const regex = new RegExp(escapedValue, 'i');
-				suggestions = suggestions.filter(item => regex.test(item.title));
-
-				if (!suggestions.length && currentProps.schema.restricted === false) {
+				const isExactMatch = suggestions.some(item =>
+					RegExp(`^${escapedValue}$`, 'i').test(item.value),
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js]: please create the regex outside of the some

File: packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js [hunk 2]
```diff
@@ -190,14 +190,19 @@ export default class MultiSelectTag extends React.Component {
...
+				);
+				suggestions = suggestions.filter(item => RegExp(escapedValue, 'i').test(item.value));
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js]: same here

File: packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js [hunk 3]
```diff
@@ -217,9 +219,11 @@ export class MultiSelectTagWidgetComponent extends React.Component {
 
 			if (filterText) {
 				const escapedValue = escapeRegexCharacters(filterText.trim());
-				const regex = new RegExp(escapedValue, 'i');
-				suggestions = suggestions.filter(item => regex.test(item.label));
-				if (suggestions.length === 0 && currentProps.schema.createIfNoneMatch) {
+				const isExactMatch = suggestions.some(item =>
+					RegExp(`^${escapedValue}$`, 'i').test(item.label),
```
Comment (jmfrancois) [inline packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js]: same here

File: packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js (hunk 1 above)
Comment (yonmey) [reply packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js]: done

File: packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js (hunk 2 above)
Comment (yonmey) [reply packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js]: done

File: packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js (hunk 3 above)
Comment (yonmey) [reply packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js]: done

File: packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js [hunk 4]
```diff
@@ -190,14 +190,18 @@ export default class MultiSelectTag extends React.Component {
...
 				const escapedValue = escapeRegexCharacters(currentValue.trim());
-				const regex = new RegExp(escapedValue, 'i');
-				suggestions = suggestions.filter(item => regex.test(item.title));
-
-				if (!suggestions.length && currentProps.schema.restricted === false) {
+				const exactMatchRx = new RegExp(`^${escapedValue}$`, 'i');
+				const similarValueRx = new RegExp(escapedValue, 'i');
+				suggestions = suggestions.filter(item => similarValueRx.test(item.value));
+				if (
+					!suggestions.some(item => exactMatchRx.test(item.value)) &&
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js]: Filter are on title, not value

File: packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js [hunk 5]
```diff
@@ -190,14 +190,18 @@ export default class MultiSelectTag extends React.Component {
...
+					!currentProps.schema.restricted &&
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js]: You changed the default restricted by changing the `=== false` test

File: packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.test.js [hunk 6]
```diff
@@ -55,16 +59,21 @@ describe('MultiSelectTag field', () => {
...
 		// then
 		expect(instance.state.suggestions).toEqual([
-			{ title: 'toto', value: 'titi' },
-			{ title: 'tata', value: 'tutu' },
+			{ title: 'TITI', value: 'titi' },
+			{ title: 'TUTU', value: 'tutu' },
 		]);
+
+		// when : trigger a props update
+		ReactDOM.render(<MultiSelectTag {...props} value={['titi', 'tutu']} />, node);
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.test.js]: Don't change the existing tests, everything should work as before.

But add the tests that check your code
- filter with one of the current values --> no new
- filter with an exact suggestion patch —> no new
- filter without suggestion or value match --> new (already covered by an existing test)

**Pull request merged**